lib/storage/keyval/leader.go

```diff
@@ -48,7 +50,17 @@ func (b *electingBackend) StepDown() {
 	b.Leader.StepDown()
 }
 
+func (b *electingBackend) CopyWithOptions(opts ...EtcdOption) storage.Backend {
+	engine := b.engine.copyWithOptions(opts...)
+	return &electingBackend{
+		Backend: b.backend,
+		Leader:  b.Leader,
+		backend: b.backend,
```

Comment:
Hmm, what's the second "backend" parameter here for? Looks like everywhere it's used it's always set to the same value as "Backend"? Or is that just to have access to the actual type, not the interface?

Reply:
Sure, I'll comment on that in code as well. The second copy of the backend so it can be manipulated since it needs access to the concrete value. I've kept the `Backend` as well since it's required for `electingBackend` to implicitly satisfy the `storage.Backend` interface.